frontend/src/metabase/metadata/pages/DataModel/components/PreviewSection/PreviewSection.tsx

```diff
@@ -12,34 +21,50 @@ import type {
 
 import { FilteringPreview } from "./FilteringPreview";
 import { ObjectDetailPreview } from "./ObjectDetailPreview";
+import S from "./PreviewSection.module.css";
 import { TablePreview } from "./TablePreview";
 import type { PreviewType } from "./types";
 import { getPreviewTypeData } from "./utils";
 
+// memoize components for smooth column resizing experience
+const MemoizedFilteringPreview = memo(FilteringPreview);
+const MemoizedObjectDetailPreview = memo(ObjectDetailPreview);
+const MemoizedTablePreview = memo(TablePreview);
```

Comment:
Why not do this where the components are defined?

Reply:
I did it like this to communicate the intent of memoization better, and also to avoid the memoized components "know" why they're memoized (because the reason they are memoized is because of the context they're used in, i.e. the parent) - but I guess this was already leaking, and they're all a part of one module anyway.

Done: 955bc5789efdd4ac0668b4ce20e77eac7eaaa56b